From 338de775796c2102ce680eaa983b719b50e9f3ee Mon Sep 17 00:00:00 2001
From: Aaron Kimball <aaron@cloudera.com>
Date: Fri, 12 Mar 2010 14:46:18 -0800
Subject: [PATCH 0052/1020] HADOOP-5469. Exposing Hadoop metrics via HTTP

Description: Implement a "/metrics" URL on the HTTP server of Hadoop daemons, to expose metrics data to users via their web browsers, in plain-text and JSON.
Reason: New feature
Author: Philip Zeyliger
Ref: UNKNOWN
---
 .../conf.pseudo/hadoop-metrics.properties          |   65 ++++++++++++++++++++
 1 files changed, 65 insertions(+), 0 deletions(-)
 create mode 100644 example-confs/conf.pseudo/hadoop-metrics.properties

diff --git a/example-confs/conf.pseudo/hadoop-metrics.properties b/example-confs/conf.pseudo/hadoop-metrics.properties
new file mode 100644
index 0000000..052b62b
--- /dev/null
+++ b/example-confs/conf.pseudo/hadoop-metrics.properties
@@ -0,0 +1,65 @@
+# Configuration of the "dfs" context for null
+#dfs.class=org.apache.hadoop.metrics.spi.NullContext
+
+# Configuration of the "dfs" context for file
+#dfs.class=org.apache.hadoop.metrics.file.FileContext
+#dfs.period=10
+#dfs.fileName=/tmp/dfsmetrics.log
+
+# Configuration of the "dfs" context for ganglia
+# dfs.class=org.apache.hadoop.metrics.ganglia.GangliaContext
+# dfs.period=10
+# dfs.servers=localhost:8649
+
+# Configuration of the "dfs" context for /metrics
+dfs.class=org.apache.hadoop.metrics.spi.NoEmitMetricsContext
+
+
+# Configuration of the "mapred" context for null
+#mapred.class=org.apache.hadoop.metrics.spi.NullContext
+
+# Configuration of the "mapred" context for /metrics
+mapred.class=org.apache.hadoop.metrics.spi.NoEmitMetricsContext
+
+# Configuration of the "mapred" context for file
+#mapred.class=org.apache.hadoop.metrics.file.FileContext
+#mapred.period=10
+#mapred.fileName=/tmp/mrmetrics.log
+
+# Configuration of the "mapred" context for ganglia
+# mapred.class=org.apache.hadoop.metrics.ganglia.GangliaContext
+# mapred.period=10
+# mapred.servers=localhost:8649
+
+
+# Configuration of the "jvm" context for null
+#jvm.class=org.apache.hadoop.metrics.spi.NullContext
+
+# Configuration of the "jvm" context for /metrics
+jvm.class=org.apache.hadoop.metrics.spi.NoEmitMetricsContext
+
+# Configuration of the "jvm" context for file
+#jvm.class=org.apache.hadoop.metrics.file.FileContext
+#jvm.period=10
+#jvm.fileName=/tmp/jvmmetrics.log
+
+# Configuration of the "jvm" context for ganglia
+# jvm.class=org.apache.hadoop.metrics.ganglia.GangliaContext
+# jvm.period=10
+# jvm.servers=localhost:8649
+
+# Configuration of the "rpc" context for null
+#rpc.class=org.apache.hadoop.metrics.spi.NullContext
+
+# Configuration of the "rpc" context for /metrics
+rpc.class=org.apache.hadoop.metrics.spi.NoEmitMetricsContext
+
+# Configuration of the "rpc" context for file
+#rpc.class=org.apache.hadoop.metrics.file.FileContext
+#rpc.period=10
+#rpc.fileName=/tmp/rpcmetrics.log
+
+# Configuration of the "rpc" context for ganglia
+# rpc.class=org.apache.hadoop.metrics.ganglia.GangliaContext
+# rpc.period=10
+# rpc.servers=localhost:8649
-- 
1.7.0.4

